cli/src/commands/new.rs

```diff
@@ -184,11 +185,18 @@ pub(crate) fn cmd_new(
 
     let mut tx = workspace_command.start_transaction();
     let merged_tree = merge_commit_trees(tx.repo(), &parent_commits)?;
-    let new_commit = tx
+    let mut commit_builder = tx
         .repo_mut()
         .new_commit(parent_commit_ids, merged_tree.id())
-        .set_description(join_message_paragraphs(&args.message_paragraphs))
-        .write()?;
+        .detach();
+    let mut description = join_message_paragraphs(&args.message_paragraphs);
+    if !description.is_empty() {
+        commit_builder.set_description(description);
+        let temp_commit = commit_builder.write_hidden()?;
+        description = add_trailers(ui, &tx, &temp_commit)?;
```

Comment:
What's the benefit to using a hidden commit here (setting aside the fact that our current trailers API doesn't support just adding trailers to a string)?

I see you did the same for `jj describe`, and I don't see any comments about it, so figured I should ask.

Reply:
Ah, right, since these are templated... sorry for the noise.